src/blockedPage/blocked.js

```diff
@@ -1,7 +1,9 @@
 document.addEventListener('DOMContentLoaded', function () {
-  const message = new URL(document.URL).searchParams.get("msg");
-  const union = new URL(document.URL).searchParams.get("union");
+  const params = new URL(document.URL).searchParams;
+  const message = params.get("msg");
+  const union = params.get("union");
+  const union_url = params.get("union_url");
+  Array.forEach(document.querySelectorAll(".union-name"), (el,i) => { el.innerText = union; });
+  document.getElementById("union-link").href = union_url;
```

Comment:
Seems like we might eventually want to switch to a better templating option, huh? But got now, this looks great.

Reply:
I agree, but I've never made a browser extension before, what would be considered minimal enough?